lib/pbench/test/functional/agent/cli/commands/tools/test_clear_tools.py

```diff
@@ -208,5 +206,151 @@ def test_clear_tools_test70(monkeypatch, agent_config, pbench_run, pbench_cfg):
+    assert fubar6_host.exists() is True
+
+
+def test_clear_tools_test74(monkeypatch, agent_config, pbench_run, pbench_cfg):
+    """ Attempt to clear all tools in default group, by name, separated by
+    comma"""
+    monkeypatch.setenv("_PBENCH_AGENT_CONFIG", str(pbench_cfg))
+    default_group = pbench_run / "tools-v1-default"
+    default_group.mkdir(parents=True)
+    fubar6_host = default_group / "fubar6.example.com"
+    fubar6_host.mkdir(parents=True, exist_ok=True)
+    pidstat6_tool = fubar6_host / "pidstat"
+    pidstat6_tool.touch()
+    mpstat6_tool = fubar6_host / "mpstat"
+    mpstat6_tool.touch()
+    iostat6_tool = fubar6_host / "iostat"
+    iostat6_tool.touch()
+
+    command = ["pbench-clear-tools", "--name=pidstat,mpstat,iostat"]
+    out, err, exitcode = pytest.helpers.capture(command)
+    assert b"" == out
+    assert pidstat6_tool.exists() is False
+    assert mpstat6_tool.exists() is False
+    assert iostat6_tool.exists() is False
+    assert fubar6_host.exists() is False
+    assert default_group.exists() is True
+
+
+def test_clear_tools_test75(monkeypatch, agent_config, pbench_run, pbench_cfg):
+    """ Attempt to clear all tools in non-default group, by name, separated by
+        comma"""
+    another_group = pbench_run / "tools-v1-another"
+    another_group.mkdir(parents=True)
+    fubar7_host = another_group / "fubar7.example.com"
+    fubar7_host.mkdir(parents=True, exist_ok=True)
+    pidstat7_tool = fubar7_host / "pidstat"
+    pidstat7_tool.touch()
+    mpstat7_tool = fubar7_host / "mpstat"
+    mpstat7_tool.touch()
+
+    command = ["pbench-clear-tools", "--group=another", "--name=pidstat,mpstat"]
+    out, err, exitcode = pytest.helpers.capture(command)
+    assert b"" == out
+    assert pidstat7_tool.exists() is False
+    assert mpstat7_tool.exists() is False
+    assert fubar7_host.exists() is False
+    assert another_group.exists() is False
+
+
+def test_clear_tools_test76(monkeypatch, agent_config, pbench_run, pbench_cfg):
+    """ Attempt to clear a group, by name, registered on multiple
+    hosts"""
+    another_group = pbench_run / "tools-v1-another"
+    another_group.mkdir(parents=True)
+
+    fubar7_1_host = another_group / "fubar7_1.example.com"
+    fubar7_1_host.mkdir(parents=True, exist_ok=True)
+    pidstat7_1_tool = fubar7_1_host / "pidstat"
+    pidstat7_1_tool.touch()
+
+    fubar7_2_host = another_group / "fubar7_2.example.com"
+    fubar7_2_host.mkdir(parents=True, exist_ok=True)
+    pidstat7_2_tool = fubar7_2_host / "pidstat"
+    pidstat7_2_tool.touch()
+
+    command = ["pbench-clear-tools", "--group=another", "--remote=fubar7_2.example.com"]
+    out, err, exitcode = pytest.helpers.capture(command)
+    assert b"" == out
+    assert pidstat7_1_tool.exists() is True
+    assert pidstat7_2_tool.exists() is False
+    assert fubar7_1_host.exists() is True
+    assert fubar7_2_host.exists() is False
+    assert another_group.exists() is True
```

Comment:
I'm a fan of "code is documentation", but I'm still inclined to boil this down to just two assertions:
- `pidstat7_1_tool` must exist
- `fubar7_2_host` must not exist
If those two hold, then they others must hold as well, so we don't need to test them.